bindings/python/pymongoarrow/api.py

```diff
@@ -233,3 +239,42 @@ def aggregate_numpy_all(collection, pipeline, *, schema, **kwargs):
     return _arrow_to_numpy(
         aggregate_arrow_all(collection, pipeline, schema=schema, **kwargs), schema
     )
+
+
+def write(collection, tabular, mode="insert"):
+    validate_schema(tabular.schema)
+    cur_batch = []
+    cur_size = 0
+    cur_offset = 0
+    from collections import Counter
+
+    results = Counter({})
+
+    def transform_bwe(bwe, offset):
+        bwe["nInserted"] += cur_offset
+        for i in bwe["writeErrors"]:
+            i["index"] += offset
+        return bwe
+
+    for row in tabular.to_pylist():
```

Comment:
to_pylist inflates the entire table to a list. Is there any way to do it iteratively?

Reply:
There's a way to iterate over the columns, but not the rows. Also, wasn't it a non-goal to deal with datasets larger than what can fit in memory?

Reply:
That's true. Changed my approach to use a generator and `to_batches` instead.